src/streams/WritableStream.ts

```diff
@@ -22,70 +18,82 @@ export interface Record<T> {
  */
 export enum State { Closed, Closing, Errored, Waiting, Writable }
 
+// This function is basically a context check to protect against calling WritableStream methods with incorrect context
+// (as one might accidentally do when passing a method as callback)
 function isWritableStream(x: any): boolean {
 	return Object.prototype.hasOwnProperty.call(x, '_underlyingSink');
 }
 
+/**
+ * The Sink interface defines the methods a module can implement to create a target sink for a `WritableStream`.
+ *
+ * The Stream API provides a consistent stream API while `ReadableStream.Source` and `WritableStream.Sink` implementors
+ * provide the logic to connect a stream to specific data sources & sinks.
+ */
 export interface Sink<T> {
 
 	/**
 	 * Indicates the stream is prematurely closing due to an error.  The sink should do any necessary cleanup
-	 * and release resources.
+	 * and release resources. When a stream calls `abort` it will discard any queued chunks. If the sink does not
+	 * provide an `abort` method then the stream will call `close` instead.
+	 *
 	 * @param reason The reason the stream is closing.
 	 */
-	abort(reason?: any): Promise<void>;
+	abort?: (reason?: any) => Promise<void>;
```

Comment:
I checked the history, and indeed I changed to that syntax when I made the optional. I tried the syntax you suggested and it does work now, so I've pushed an update with that change.